### ISSUE TITLE
What's to become of Caret?

### ISSUE THREAD

**Issue by prairiedock** [thomaswilburn/Caret#719]
### Type of issue (choose one):
  - [ ] Bug
  - [ ] Feature request
  - [ x] Question

### Description

Since Google has provided a sunset date for all Chrome apps, does this mean Caret is now on the executioner's block?

That would be a terrible shame, since Caret is a more-than-worthy text editor, highly stable and bug-free in my experience. It's been my go-to editor for 5 years now, and has no real competition that I'm aware of.

Is it possible to present Caret as an extension instead of an app?

I don't know what Google is thinking here. What are they going to do about the Secure Shell app, or the utterly essential Chrome OS Recovery app?

### System details

Chrome OS, stable channel, dev mode.
Chrome v. 79.0.3945.123

*Please provide your Chrome version and operating system here.*

**Comment by thomaswilburn** [comment]
There's some movement toward implementing the native filesystem APIs, which would let Caret run as a progressive web app. I don't have any plans to do a PWA version myself until multiple browsers have that built in. I don't know if that's worth it--it seems to me that since I first wrote Caret, tools like Glitch have really changed the landscape.

**Comment by prairiedock** [comment]
Wouldn't a PWA be usable online-only? (Then I don't think I'm interested.)

If you don't mind my asking, what do you use (or plan to use) for your own
coding and text editing needs? I've already scrambled to get SciTE working
on Crostini (I'd already put some effort into a SciTEUser.properties file
years ago on Windows), but that's not as smooth an experience to use as
Caret. But it is configurable via Lua, which is a plus, and something Caret
doesn't have, so....

--prairiedock

**Comment by thomaswilburn** [comment]
You'd need to visit it once to install it, and then it would work offline. The UI would be challenging--it would run in a browser tab, there's a lot of questions that I think Google still doesn't have a real answer to.

I don't really work on Chrome OS very much anymore--I went back to an old Linux laptop, so I'm pretty much just running Sublime everywhere. A lot of the appeal of Chrome apps to me initially was that they were supported anywhere that Chrome is. When they killed the store on Windows (and announced EOL for the non-Chrome OS APIs), it was pretty clear I would need to find another solution. Mostly I support Caret now (to the extent I do) because it's still used in a number of schools to teach coding to kids.

**Comment by prairiedock** [comment]
Thanks for this.

Mostly I've avoided (straight) linux because of (what I've perceived as)
the continual parade of device driver-, audio and bluetooth-, suspend and
power management-, and other hardware-issues. I experienced these myself
when I tried to install it about a decade ago, and they can be
disheartening.

Some sources I read contend that this is no longer the case, but other
knowledgeable sources still complain constantly about the bugs. Chrome OS,
OTOH, has been amazingly solid, and I'm done with Windows.

Thanks for Caret these last years!
